### DISCUSSION TITLE
Can't Generate Links Anymore - Error 429 "Too Many Requests"

### DISCUSSION THREAD

**Discussion by CMAYK** [s0hv/foo_discord_rich#23] [Q&A] [answered]
I started using this version of foo_discord_rich, and it was initially working great for the first few days. However, I noticed recently that I can no longer upload any images, and I have to manually insert URLs.

I tried pausing foobar for a day to see if the ratelimit would refresh, but I still can't get it to generate new links.

Is there a way to get the ratelimit to refresh? Is there something in the background keeping it from working properly?
Did Imgur change something?
My guess is that I am using the plugin wrong, but many of my friends are encountering the same issue and error as me. 

Here is a snippet from my console:

> Discord Rich Presence Integration v1.3.0-beta-6+4aec8f6: artwork uploader exited with status: 1 and error: Error while parsing imgur response from {"errors":[{"id":"legacy-api-5df468c8bc-qh7vf/7wFFxB3x44-3205233","code":"429","status":"Too Many Requests","detail":"Too Many Requests"}]}
> : missing field `data` at line 1 column 139
> x-ratelimit-clientremaining: 0
> x-ratelimit-userremaining: 499
> x-ratelimit-userlimit: 500
> Discord Rich Presence Integration v1.3.0-beta-6+4aec8f6: artwork uploader exited with status: 1 and error: Error while parsing imgur response from 
> <?xml version="1.0" encoding="utf-8"?>
> <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
>  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
> <html>
>   <head>
>     <title>503 client read error</title>
>   </head>
>   <body>
>     <h1>Error 503 client read error</h1>
>     <p>client read error</p>
>     <h3>Error 54113</h3>
>     <p>Details: cache-lga21953-LGA 1714852183 3748740353</p>
>     <hr>
>     <p>Varnish cache server</p>
>   </body>
> </html>
> : expected value at line 2 column 1
> Discord Rich Presence Integration v1.3.0-beta-6+4aec8f6: 0 entries updated

**Comment by s0hv** [comment, marked as answer]
I recommend you just compile it yourself with your own API key. Currently imgur just breaks when the API key is shared by multiple people and imgur support has not responded to my ticket.

**Reply by CMAYK** [reply]
Didn't even cross my mind, thank you for the quick reply
